### DISCUSSION TITLE
tfsec compliance checks.

### DISCUSSION THREAD

**Discussion by raypettersen** [aquasecurity/trivy#5786] [Q&A] [answered]
### Question

I'm currently stuck with some compliance checks that I'd like to run with trivy (previously tfsec). One of these json checks looks like:
```
custom_aws_region_check_tfchecks.json
{
  "checks": [
    {
      "code": "CUSTOM_AWS_REGION",
      "description": "Ensure AWS region is set to eu-north-1",
      "requiredTypes": ["provider"],
      "requiredLabels": ["aws"],
      "severity": "HIGH",
      "errorMessage": "AWS provider region is not set to eu-north-1",
      "impact": "Non-compliance with region requirements",
      "resolution": "Set the AWS region to eu-north-1",
      "matchSpec": {
        "name": "region",
        "action": "equals",
        "value": "eu-north-1"
      },
      "relatedLinks": []
    }
  ]
}
```

And in tfsec we're exporting the result as junit xml, which is picked up by gitlab.

I understand that it's currently a transition from tfsec to trivy, but has this functionality been ported? If not, will it ever be ported?

### Target

None

### Scanner

Misconfiguration

### Output Format

None

### Mode

None

**Comment by nikpivkin** [comment, marked as answer]
Hi @raypettersen !

Regarding the first question. 
This functionality is not ported. Custom policies are now written using [Rego](https://aquasecurity.github.io/trivy/v0.48/docs/scanner/misconfiguration/custom/). 

You can [convert](https://aquasecurity.github.io/trivy/v0.48/docs/configuration/reporting/#junit) the scan result to a GitLab report using the [template](https://github.com/aquasecurity/trivy/blob/main/contrib/junit.tpl). 
`$ trivy image --format template --template "@contrib/junit.tpl" -o junit-report.xml  golang:1.12-alpine`

/cc @itaysk
